src/equations/maxwell_1d.jl

```diff
@@ -0,0 +1,106 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+@doc raw"""
+    MaxwellEquations1D
+
+The Maxwell equations of electro dynamics
+```math
+\frac{\partial}{\partial t}
+\begin{pmatrix}
+E \\ B
+\end{pmatrix}
++ 
+\frac{\partial}{\partial x}
+\begin{pmatrix}
+c^2 B \\ E
+\end{pmatrix}
+=
+\begin{pmatrix}
+0 \\ 0 
+\end{pmatrix}
+```
+in one dimension with speed of light `c = 299792458 m/s` (in vacuum).
+In one dimension the Maxwell equations reduce to a wave equation.
+The orthogonal magnetic (e.g.`B_y`) and electric field (`E_z`) propagate as waves 
+through the domain in `x`-direction.
+For reference, see 
+- e.g. p.15 of Numerical Methods for Conservation Laws: From Analysis to Algorithms 
+  https://doi.org/10.1137/1.9781611975109
+
+- or equation (1) in https://inria.hal.science/hal-01720293/document
+"""
+struct MaxwellEquations1D{RealT <: Real} <:
+       AbstractLinearScalarAdvectionEquation{1, 2}
+    speed_of_light::SVector{1, RealT} # c
+end
+
+function MaxwellEquations1D(c::Real = 299792458)
+    MaxwellEquations1D(SVector(c))
+end
```

Comment:
The default value should be a `Float64`, shouldn't it?
```suggestion
function MaxwellEquations1D(c::Real = 299_792_458.0)
    MaxwellEquations1D(SVector(c))
end
```